### PR TITLE
feat(openapi): adapt OpenAPIv2 "List draft data" and "List commit data"

### PR DESCRIPTION
**Pull request by wangyuqing0424**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-27k14n3 OpenAPIv2 对接 —— ListData](https://app.clickup.com/t/27k14n3)

File: graviti/openapi/data.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
```
Comment (graczhual) [inline graviti/openapi/data.py:1]: adapt OpenAPI list_draft_data and list_commit_data

File: graviti/openapi/data.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    dataset: str,
+    *,
+    draft_number: int,
+    sheet: str,
+    columns: Optional[str] = None,
+    order_by: Optional[str] = None,
+    offset: int = 0,
+    limit: int = 128,
+) -> Dict[str, Any]:
+    """Execute the OpenAPI`GET /v2/datasets/{owner}/{dataset}/drafts/{draft_number}\
```
Comment (graczhual) [inline graviti/openapi/data.py:26]: add space after OpenAPI

File: graviti/openapi/data.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        The response of OpenAPI.
+
+    Examples:
+        >>> list_draft_data(
+        ...     "https://api.graviti.com/",
+        ...     "ACCESSKEY-********",
+        ...     "czhual",
+        ...     "MNIST",
+        ...     1,
+        ...     "train",
```
Comment (graczhual) [inline graviti/openapi/data.py:53]: draft_number=, sheet=

File: graviti/openapi/data.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    Arguments:
+        url: The URL of the graviti website.
+        access_key: User's access key.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        columns: The string of column names separated by commas.
+            Multiple indexes can be expressed using dots. None means to get all columns.
+        order_by: The string of column names separated by commas.
```
Comment (linjiX) [inline graviti/openapi/data.py:38]: do we have defined the behavior of `order_by`? @graczhual

File: graviti/openapi/data.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            ],
+            "offset": 0,
+            "record_size": 128,
+            "total_count": 70000
+        }
+    """
+    url = urljoin(url, f"v2/datasets/{owner}/{dataset}/drafts/{draft_number}/sheets/{sheet}/data")
+    params: Dict[str, Any] = {"offset": offset, "limit": limit}
+
+    if columns:
```
Comment (graczhual) [inline graviti/openapi/data.py:107]: is not None

File: graviti/openapi/data.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    /sheets/{sheet}/data`.
+
+    Arguments:
+        url: The URL of the graviti website.
+        access_key: User's access key.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        columns: The string of column names separated by commas.
```
Comment (linjiX) [inline graviti/openapi/data.py:36]: comma is the separator?  what if there is comma in the column name？

File: graviti/openapi/data.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    return open_api_do("GET", access_key, url, params=params).json()  # type: ignore[no-any-return]
+
+
+def list_commit_data(
+    url: str,
+    access_key: str,
+    owner: str,
+    dataset: str,
+    *,
+    commit_id: int,
```
Comment (graczhual) [inline graviti/openapi/data.py:121]: str

File: graviti/openapi/data.py [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+        commit_id: The commit id.
+        sheet: The sheet name.
+        columns: The string of column names separated by commas.
+            Multiple indexes can be expressed using dots. None means to get all columns.
+        order_by: The string of column names separated by commas.
+            Earlier sorting has higher priority
+        offset: The offset of the page.
+        limit: The limit of the page.
+
+
```
Comment (graczhual) [inline graviti/openapi/data.py:145]: delete one line

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+#
+
+"""Interfaces about the data."""
+from typing import Any, Dict, Optional
+from urllib.parse import urljoin
+
+from graviti.openapi.requests import open_api_do
+
+
+def list_draft_data(
```
Comment (linjiX) [inline graviti/openapi/data.py:13]: `list_draft_data` and `list_commit_data` is almost the same, can we decrease the duplicate code？

File: graviti/openapi/data.py (hunk 4 above)
Comment (graczhual) [reply graviti/openapi/data.py:38]: ben add the description about order_by in the doc, here the translation here is not very accurate

File: graviti/openapi/data.py [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+        ...     "https://api.graviti.com/",
+        ...     "ACCESSKEY-********",
+        ...     "czhual",
+        ...     "MNIST",
+        ...      1,
+        ...     "train",
+        ... )
+
+    """
+    url = urljoin(url, f"v2/datasets/{owner}/{dataset}/dcommits/{commit_id}/sheets/{sheet}/data")
```
Comment (graczhual) [inline graviti/openapi/data.py:160]: commits not dcommits

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py (hunk 9 above)
Comment (graczhual) [reply graviti/openapi/data.py:13]: Here, we can not modify the arguments. The duplicate code is the judgment on whether the argument is None or "". It doesn't feel necessary to merge them. Almost every API has this code. Or we need to put this judgement to open-api-do?

File: graviti/openapi/data.py [hunk 11]
```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
```
Comment (graczhual) [inline graviti/openapi/data.py:3]: adapt OpenAPIv2 "List draft data" and "List commit data"

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+    Arguments:
+        url: The URL of the graviti website.
+        access_key: User's access key.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        commit_id: The commit id.
+        sheet: The sheet name.
+        columns: The string of column names separated by commas.
+            Multiple indexes can be expressed using dots. None means to get all columns.
+        order_by: The string of column names separated by commas.
```
Comment (graczhual) [inline graviti/openapi/data.py:140]: do not need the first sentence here

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py [hunk 13]
```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Interfaces about the data."""
```
Comment (graczhual) [inline graviti/openapi/data.py:6]: add a blank line here

File: graviti/openapi/data.py [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+    Arguments:
+        url: The URL of the graviti website.
+        access_key: User's access key.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        draft_number: The draft number.
+        sheet: The sheet name.
+        columns: The string of column names separated by commas.
+            Multiple indexes can be expressed using dots. None means to get all columns.
+        order_by: The string determine the order of sorting by the order in \
```
Comment (graczhual) [inline graviti/openapi/data.py:38]: this line use \, but columns do not use

File: graviti/openapi/data.py [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+
+    """
+    url = urljoin(url, f"v2/datasets/{owner}/{dataset}/commits/{commit_id}/sheets/{sheet}/data")
+
+    return _list_data(
+        access_key, url, columns=columns, order_by=order_by, offset=offset, limit=limit
+    )
+
+
+def _list_data(
```
Comment (graczhual) [inline graviti/openapi/data.py:162]: private functions move to the front

File: graviti/openapi/data.py [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+    access_key: str,
+    url: str,
+    *,
+    columns: Optional[str] = None,
+    order_by: Optional[str] = None,
+    offset: int = 0,
+    limit: int = 128,
```
Comment (graczhual) [inline graviti/openapi/data.py:169]: do not need default value here

File: graviti/openapi/data.py [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+    Returns:
+        The response of OpenAPI.
+
+    Examples:
+        >>> list_commit_data(
+        ...     "https://api.graviti.com/",
+        ...     "ACCESSKEY-********",
+        ...     "czhual",
+        ...     "MNIST",
+        ...     commit_id = "1",
```
Comment (graczhual) [inline graviti/openapi/data.py:150]: use a correct commit id like "fde63f357daf46088639e9f57fd81cad"

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/data.py (hunk 14 above)
Comment (wangyuqing0424) [reply graviti/openapi/data.py:38]: Just because it would be too long in one line but colums is ok

File: graviti/openapi/data.py [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+    order_by: Optional[str] = None,
+    offset: int = 0,
+    limit: int = 128,
+) -> Dict[str, Any]:
+    """Execute the OpenAPI `GET /v2/datasets/{owner}/{dataset}/drafts/{draft_number}\
+    /sheets/{sheet}/data`.
+
+    Arguments:
+        url: The URL of the graviti website.
+        access_key: User's access key.
```
Comment (graczhual) [inline graviti/openapi/data.py:51]: Accesskey url modify the order

File: graviti/openapi/data.py [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+        offset: The offset of the page.
+        limit: The limit of the page.
+
+    Returns:
+        The response of OpenAPI.
+
+    Examples:
+        >>> list_draft_data(
+        ...     "https://api.graviti.com/",
+        ...     "ACCESSKEY-********",
```
Comment (graczhual) [inline graviti/openapi/data.py:69]: accesskey, url, change the order

**Review by graczhual — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

**Pull request merged**